packages/vm/src/wasm_backend/engine.rs

```diff
@@ -16,14 +16,31 @@ use super::limiting_tunables::LimitingTunables;
 /// https://github.com/WebAssembly/memory64/blob/master/proposals/memory64/Overview.md
 const MAX_WASM_PAGES: u32 = 65536;
 
-fn cost(_operator: &Operator) -> u64 {
+fn cost(operator: &Operator) -> u64 {
     // A flat fee for each operation
     // The target is 1 Teragas per second (see GAS.md).
     //
     // In https://github.com/CosmWasm/cosmwasm/pull/1042 a profiler is developed to
     // identify runtime differences between different Wasm operation, but this is not yet
     // precise enough to derive insights from it.
-    170
+    //
+    // Please note that any changes to this function need to be accompanied by a bump of
+    // `MODULE_SERIALIZATION_VERSION` to avoid cached modules from using different amounts of gas
+    // compared to newly compiled ones.
```

Comment:
Nice!

Now I was wondering if/how we can turn this function into something we can hash and add to the caching path. Either by serializing the function (some scripting language) or serialize the op -> cost as a table/map. Could you create a ticket for that as mid or long-term improvement?

Reply:
I was also wondering if we can test the cache compatibility by having a test with a cache folder with one wasm + module file inside and testing that it can be executed and works as expected (same gas, etc.).
That would probably catch some panics on the wasmer side and if we assert gas, also gas cost changes.